frontend/components/Spinner/Spinner.tsx

```diff
@@ -1,17 +1,36 @@
 import React from "react";
 import classnames from "classnames";
 
+type Size = "x-small" | "small" | "medium";
+
 interface ISpinnerProps {
   small?: boolean;
   button?: boolean;
   white?: boolean;
+  /** The size of the spinner. defaults: `"medium"` */
+  size?: Size;
+  /** Include the background container styling for the spinner. defaults: `true` */
+  includeContainer?: boolean;
+  /** Center the spinner in its parent. defaults: `true` */
+  centered?: boolean;
+  className?: string;
 }
 
-const Spinner = ({ small, button, white }: ISpinnerProps): JSX.Element => {
-  const classOptions = classnames(`loading-spinner`, {
+const Spinner = ({
+  small,
+  button,
+  white,
```

Comment:
I liked your ideas about fixing these to classes. But def can be done on another PR @ghernandez345

Reply:
for sure. i'll create a ticket for it

Reply:
created it here https://github.com/fleetdm/fleet/issues/8142